spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -10,10 +18,32 @@
     end
 
     context "when sending a welcome_thread notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has commented in a welcome thread"
+      let(:welcome_thread_article) { create(:article, title: "Welcome Thread") }
+      let(:welcome_thread_comment) { create(:comment, commentable: welcome_thread_article, user: user) }
+      let(:article) { create(:article) }
+      let(:comment) { create(:comment, commentable: article, user: receiving_user) }
+
+      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
+        expect(receiving_user.notifications.count).to eq(0)
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+
+        expect(receiving_user.notifications.count).to eq(1)
+        expect(receiving_user.notifications.first.notifiable).to eq(welcome_broadcast)
+      end
+
+      it "does not send a notification to a user who has commented in a welcome thread", elasticsearch: true do
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
```

Comment:
Why do we use `receiving_user` here, but then on line 27, we check `user`? Seems like this test isn't testing the right thing here

Reply:
I've made this fix! 👍